tests/Language/PureScript/Ide/SourceFileSpec.hs

```diff
@@ -16,55 +16,62 @@ span0 = P.SourceSpan "ModuleLevel" (P.SourcePos 0 0) (P.SourcePos 1 1)
 span1 = P.SourceSpan "" (P.SourcePos 1 1) (P.SourcePos 2 2)
 span2 = P.SourceSpan "" (P.SourcePos 2 2) (P.SourcePos 3 3)
 
+ann0, ann1, ann2 :: P.SourceAnn
+ann0 = (span0, [])
+ann1 = (span1, [])
+ann2 = (span2, [])
+
 typeAnnotation1, value1, synonym1, class1, class2, data1, data2, valueFixity, typeFixity, foreign1, foreign2, foreign3, member1 :: P.Declaration
-typeAnnotation1 = P.TypeDeclaration (P.Ident "value1") P.REmpty
-value1 = P.ValueDeclaration (P.Ident "value1") P.Public [] []
-synonym1 = P.TypeSynonymDeclaration (P.ProperName "Synonym1") [] P.REmpty
-class1 = P.TypeClassDeclaration (P.ProperName "Class1") [] [] [] []
-class2 = P.TypeClassDeclaration (P.ProperName "Class2") [] [] []
-  [P.PositionedDeclaration span2 [] member1]
-data1 = P.DataDeclaration P.Newtype (P.ProperName "Data1") [] []
-data2 = P.DataDeclaration P.Data (P.ProperName "Data2") [] [(P.ProperName "Cons1", [])]
+typeAnnotation1 = P.TypeDeclaration ann1 (P.Ident "value1") P.REmpty
+value1 = P.ValueDeclaration ann1 (P.Ident "value1") P.Public [] []
+synonym1 = P.TypeSynonymDeclaration ann1 (P.ProperName "Synonym1") [] P.REmpty
+class1 = P.TypeClassDeclaration ann1 (P.ProperName "Class1") [] [] [] []
+class2 = P.TypeClassDeclaration ann1 (P.ProperName "Class2") [] [] [] [member1]
+data1 = P.DataDeclaration ann1 P.Newtype (P.ProperName "Data1") [] []
+data2 = P.DataDeclaration ann1 P.Data (P.ProperName "Data2") [] [(P.ProperName "Cons1", [])]
 valueFixity =
   P.ValueFixityDeclaration
+    ann1
     (P.Fixity P.Infix 0)
     (P.Qualified Nothing (Left (P.Ident "")))
     (P.OpName "<$>")
 typeFixity =
   P.TypeFixityDeclaration
+    ann1
     (P.Fixity P.Infix 0)
     (P.Qualified Nothing (P.ProperName ""))
     (P.OpName "~>")
-foreign1 = P.ExternDeclaration (P.Ident "foreign1") P.REmpty
-foreign2 = P.ExternDataDeclaration (P.ProperName "Foreign2") P.kindType
-foreign3 = P.ExternKindDeclaration (P.ProperName "Foreign3")
-member1 = P.TypeDeclaration (P.Ident "member1") P.REmpty
+foreign1 = P.ExternDeclaration ann1 (P.Ident "foreign1") P.REmpty
+foreign2 = P.ExternDataDeclaration ann1 (P.ProperName "Foreign2") P.kindType
+foreign3 = P.ExternKindDeclaration ann1 (P.ProperName "Foreign3")
+member1 = P.TypeDeclaration ann2 (P.Ident "member1") P.REmpty
 
 spec :: Spec
 spec = do
   describe "Extracting Spans" $ do
+    --TODO-gb: is this worth testing now? the spans part not so much, but returning the appropriate IdeNSValue/IdeNSType is useful perhaps?
```

Comment:
These tests have caught the stupidest mistakes before, so I'll say they're worth it. At the very least they make sure all the pattern matches are exhaustive.